### PR TITLE
Scarb Doc: cross item linkage

### PR DESCRIPTION
**Pull request by wawel37**
Waiting for [this](https://github.com/starkware-libs/cairo/pull/6522) to be merged.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -35,30 +35,30 @@ anyhow = "1"
...
+cairo-lang-sierra = { path = "../cairo/crates/cairo-lang-sierra" }
+cairo-lang-sierra-generator = { path = "../cairo/crates/cairo-lang-sierra-generator" }
+cairo-lang-sierra-to-casm = { path = "../cairo/crates/cairo-lang-sierra-to-casm" }
+cairo-lang-sierra-type-size = { path = "../cairo/crates/cairo-lang-sierra-type-size" }
+cairo-lang-starknet = { path = "../cairo/crates/cairo-lang-starknet" }
+cairo-lang-starknet-classes = { path = "../cairo/crates/cairo-lang-starknet-classes" }
+cairo-lang-syntax = { path = "../cairo/crates/cairo-lang-syntax" }
+cairo-lang-test-plugin = { path = "../cairo/crates/cairo-lang-test-plugin" }
+cairo-lang-test-runner = { path = "../cairo/crates/cairo-lang-test-runner" }
+cairo-lang-utils = { path = "../cairo/crates/cairo-lang-utils" }
```
Comment (wawel37) [inline Cargo.toml:61]: This will get changed when changes are merged on the Cairo side

**Comment by wawel37** [comment]
> Waiting for [this](https://github.com/starkware-libs/cairo/pull/6522) to be merged.

Review ready

File: extensions/scarb-doc/src/types.rs [hunk 2]
```diff
@@ -378,6 +409,80 @@ impl Module {
...
+            ids.insert(trait_.item_data.id, trait_.item_data.clone());
+            trait_.trait_constants.iter().for_each(|item| {
+                ids.insert(item.item_data.id, item.item_data.clone());
+            });
+            trait_.trait_functions.iter().for_each(|item| {
+                ids.insert(item.item_data.id, item.item_data.clone());
+            });
+            trait_.trait_types.iter().for_each(|item| {
+                ids.insert(item.item_data.id, item.item_data.clone());
+            });
```
Comment (maciektr) [inline extensions/scarb-doc/src/types.rs:464]: Maybe those types should also implement `get_all_item_ids`, so we do not need to remember to update this function whenever they change.

**Review by maciektr — COMMENTED** [review]
Not sure if it's only me or not, but passing this very specific `included_items` everywhere just bothers me a bit 😅 Maybe we can at least hide it behind some abstraction? Something like this is coming to my mind, probably not ideal though: https://gist.github.com/2028017ce043ca137530cd0281fdf55b.git
LMKWYT

**Comment by wawel37** [comment]
> Not sure if it's only me or not, but passing this very specific `included_items` everywhere just bothers me a bit 😅 Maybe we can at least hide it behind some abstraction? Something like this is coming to my mind, probably not ideal though: https://gist.github.com/2028017ce043ca137530cd0281fdf55b.git LMKWYT

I totally agree. Moved the included items to the context. Tbh I don't see any more elegant solve for this problem here, as we need to keep track of those included items throughout creating types and later generating docs. Maybe in the future we can make some refactor, which would keep the context in mind

**Review by maciektr — APPROVED** [review]
💪

**Pull request merged**